kernel/names.ml

```diff
@@ -870,7 +870,7 @@ struct
       if b = b' then Constant.SyntacticOrd.compare c c' else -1
     let equal (c, b as x) (c', b' as x') =
       x == x' || b = b' && Constant.SyntacticOrd.equal c c'
-    let hash (c, b) = (if b then 0 else 1) + Constant.SyntacticOrd.hash c
+    let hash (c, b : t) = (if b then 0 else 1) + Constant.SyntacticOrd.hash c
   end
```

Comment:
Unused type t
Looking at it more I should just remove Projection.SyntacticOrd.t since it doesn't appear in the mli and is the same as Projection.t